topostats/grains.py

```diff
@@ -742,3 +795,741 @@ def keep_largest_labelled_region(
+        # Convert to binary in case there are multiple hits in the same pixel. We dont want to have 2s, 3s etc
+        # because this would cause issues in labelling and cause grains in grains.
+        return np.sum(grain_mask_tensor[:, :, 1:], axis=-1).astype(bool)
+
+    @staticmethod
+    def get_multi_class_grain_bounding_boxes(grain_mask_tensor: npt.NDArray) -> dict:
+        """
+        Get the bounding boxes for each grain in a multi-class image tensor.
+
+        Finds the bounding boxes for each grain in a multi-class image tensor. Grains can span multiple classes, so the
+        bounding boxes are found for the combined binary mask of contiguous grains across all classes.
+
+        Parameters
+        ----------
+        grain_mask_tensor : npt.NDArray
+            3-D Numpy array of grain mask tensor.
+
+        Returns
+        -------
+        dict
+            Dictionary of bounding boxes indexed by grain number.
+        """
+        # Get the flattened mask
+        flattened_mask = Grains.flatten_multi_class_tensor(grain_mask_tensor)
+        # Label the regions
+        labelled_regions = Grains.label_regions(flattened_mask)
+        # Get the region properties
+        region_properties = Grains.get_region_properties(labelled_regions)
+        # Get the bounding boxes
+        bounding_boxes = {index: region.bbox for index, region in enumerate(region_properties)}
+        # Pad the bounding boxes
+        return {
+            index: pad_bounding_box(
+                crop_min_row=bbox[0],
+                crop_min_col=bbox[1],
+                crop_max_row=bbox[2],
+                crop_max_col=bbox[3],
+                image_shape=(grain_mask_tensor.shape[0], grain_mask_tensor.shape[1]),
+                padding=1,
+            )
+            for index, bbox in bounding_boxes.items()
+        }
+
+    @staticmethod
+    def update_background_class(
+        grain_mask_tensor: npt.NDArray,
+    ) -> npt.NDArray:
+        """
+        Update the background class to reflect the other classes.
+
+        Parameters
+        ----------
+        grain_mask_tensor : npt.NDArray
+            3-D Numpy array of the grain mask tensor.
+
+        Returns
+        -------
+        npt.NDArray
+            3-D Numpy array of image tensor with updated background class.
+        """
+        # Get the flattened mask and invert it
+        flattened_mask = Grains.flatten_multi_class_tensor(grain_mask_tensor)
+        new_background = np.where(flattened_mask == 0, 1, 0)
+        # Update the background class
+        grain_mask_tensor[:, :, 0] = new_background
+        return grain_mask_tensor.astype(bool)
+
+    @staticmethod
+    def vet_class_sizes(
+        grain_mask_tensor: npt.NDArray,
+        pixel_to_nm_scaling: float,
+        class_size_thresholds: list[list[int, int, int]],
+    ) -> npt.NDArray:
+        """
+        Vet the sizes of the classes in an image tensor.
```

Comment:
I think "Vet" is a poor description of what this method is doing.

```suggestion
        Remove grains from mask tensor below a given threshold.
```

Reply:
I've attempted to remedy the obscure descriptions, see what you think now, happy to iterate again.